### PR TITLE
CI with travis and beaver

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
FYI @sociomantic-tsunami/core-team

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
+    matrix:
+        - DC=dmd1 DIST=xenial F=production
+        - DC=dmd1 DIST=xenial F=devel
+        - DC=dmd-transitional DIST=xenial F=production
+        - DC=dmd-transitional DIST=xenial F=devel
+
+install: beaver install
+
+script: BEAVER_DOCKER_VARS="DC DIST" beaver run ci/travis.sh
```
Comment (leandro-lucarella-sociomantic) [inline .travis.yml:29]: `DIST` is automatically exported, only `DC` must be specified.

File: ci/travis.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+set -xe
+
+# Travis likes to fetch submodules recursively, but the build system assumes
+# shalow submodules fetching, so we need to remove all the recursive submodules
+# before proceeding
+git submodule foreach --recursive git submodule deinit --force --all
```
Comment (leandro-lucarella-sociomantic) [inline ci/travis.sh:7]: This should be killed, is not necessary with the stuff in the travis.yml.

File: ci/travis.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# Defaults (in case they are not set by the CI)
+F=${F:-production}
+DC=${DC:-dmd1}
+DIST=${DIST:-xenial}
```
Comment (leandro-lucarella-sociomantic) [inline ci/travis.sh:12]: Also, these defaults could be killed, as this is always expected to be run by travis, we can assume all variables will be defined (and if not it should fail).

**Pull request merged**